core/src/main/java/io/kestra/core/runners/ExecutionDelay.java

```diff
@@ -25,8 +24,24 @@ public class ExecutionDelay {
 
     @NotNull State.Type state;
 
+    @NotNull DelayType delayType;
+
     @JsonIgnore
     public String uid() {
         return String.join("_", executionId, taskRunId);
     }
+
+    /**
+     * For previous version, return RESUME_FLOW by default as it was the only case
+     * @return DelayType representing the action to do when
+     */
+    public DelayType getDelayType() {
+        return delayType == null ? DelayType.RESUME_FLOW : delayType;
+    }
+
+    public enum DelayType {
+        RESUME_FLOW,
+        RESTART_FAILED_TASK,
```

Comment:
```suggestion
        RESTART_FAILED_TASKS,
```
I assume it would restart all failed tasks, so better plural

Reply:
It only restart a specific failed task